lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -611,24 +651,74 @@ def _is_tls_fully_configured(self) -> bool:
         if not unit_http_secrets or not unit_http_secrets.get("cert"):
             return False
 
-        return self._are_all_tls_resources_stored()
+        stored = self._are_all_tls_resources_stored()
+        if stored:
+            self.peers_data.put(Scope.UNIT, "tls_configured", True)
+
+        return stored
+
+    def is_every_unit_marked_as_started(self) -> bool:
+        """Check if every unit in the cluster is marked as started."""
+        rel = self.model.get_relation(PeerRelationName)
+        for unit in rel.units.union({self.unit}):
+            if rel.data[unit].get("started") != "True":
+                return False
+        return True
+
+    def is_tls_full_configured_in_cluster(self) -> bool:
+        """Check if TLS is configured in all the units of the current cluster."""
+        rel = self.model.get_relation(PeerRelationName)
+        for unit in rel.units.union({self.unit}):
+            if rel.data[unit].get("tls_configured") != "True":
+                return False
+        return True
+
+    def is_admin_user_configured(self) -> bool:
+        """Check if admin user configured."""
+        # In case the initialisation of the admin user is not finished yet
+        return self.peers_data.get(Scope.APP, "admin_user_initialized", False)
```

Comment:
The initialization of the the admin user is a one time operation in the lifecycle of the charm - on the very first "leader_elected" hook execution - before the daemon starts, as it must be persisted on a config file on disk. 
This flag does not vary with time, it's false until it is true forever.
It is also the only reliable way to check on the first start of the charm - as no API call can be made yet

Reply:
Do I get it right: we set the flag, so that later on we can add corresp data to the config file to persist?
(Isn't it redundant to keep the databag flag after? Doesn't quite matter for the PR, I'm just curious :-) )